### PR TITLE
Nrf cleanup

### PR DESCRIPTION
**Pull request by arturo182**
This implements the changes discussed in #591. I removed the nRF51 board and some nRF52 boards that we don't want to support. Also some additional cleaning on the way.

Status

| Board  | SD | SD ver | Builds | Boots |
|-|-|-|-|-|
|feather52|no SD||yes|yes|
|feather52|s132|2.0.1|yes|yes|
|feather52|s132|5.0.0|yes|yes|
|pca10056|no SD||yes|yes|
|pca10056|s140|6.0.0|yes|yes|

Once I get the feather52 I can test the three builds for it to verify they work, but I think the code review could be started now so when I verify, we can merge.

In the future I want to get pca10040 and the Thingy52 working, but would like to merge #589 first since it changes the `board.c` files.

### CODE REVIEW

File: ports/nrf/README.md [hunk 1]
```diff
@@ -10,50 +10,41 @@ This is a port of MicroPython to the Nordic Semiconductor nRF series of chips.
...
 
-    git clone <URL>.git micropython
-    cd micropython
+    git clone <URL>.git circuitpython
+    cd circuitpython
     git submodule update --init
     make -C mpy-cross
 
-By default, the PCA10040 (nrf52832) is used as compile target. To build and flash issue the following command inside the ports/nrf/ folder:
+By default, the PCA10040 (nRF52832) is used as compile target. To build and flash issue the following command inside the ports/nrf/ folder:
```
Comment (microbuilder) [inline ports/nrf/README.md]: The default is actually `feather52` now, with S132v2.0.1 and BLE enabled by default.

**Comment by jerryneedell** [comment]
hmm -I downloaded this PR to master and built and installed the feather52 but now I cannot access the board via ampy (via USB_ as in the past.
```jerryneedell@Ubuntu-Macmini:~/projects/feather52$ ampy -d 1.5 -p /dev/ttyUSB0 ls
Traceback (most recent call last):
  File "/usr/local/bin/ampy", line 9, in <module>
    load_entry_point('adafruit-ampy==1.0.3', 'console_scripts', 'ampy')()
  File "/usr/local/lib/python3.5/dist-packages/click-6.7-py3.5.egg/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click-6.7-py3.5.egg/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/dist-packages/click-6.7-py3.5.egg/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.5/dist-packages/click-6.7-py3.5.egg/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/dist-packages/click-6.7-py3.5.egg/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/adafruit_ampy-1.0.3-py3.5.egg/ampy/cli.py", line 147, in ls
  File "/usr/local/lib/python3.5/dist-packages/adafruit_ampy-1.0.3-py3.5.egg/ampy/files.py", line 103, in ls
  File "/usr/local/lib/python3.5/dist-packages/adafruit_ampy-1.0.3-py3.5.egg/ampy/files.py", line 96, in ls
  File "/usr/local/lib/python3.5/dist-packages/adafruit_ampy-1.0.3-py3.5.egg/ampy/pyboard.py", line 265, in exec_
ampy.pyboard.PyboardError: ('exception', b'', b'Traceback (most recent call last):\r\n  File "<stdin>", line 6, in <module>\r\nOSError: 19\r\n')
jerryneedell@Ubuntu-Macmini:~/projects/feather52$ 
```

**Comment by jerryneedell** [comment]
hmmm - this may be why?
```
Adafruit CircuitPython 3.0.0-alpha.1-120-g381f47b on 2018-02-08; Bluefruit nRF52 Feather with NRF52832
>>> 
>>> 
>>> import os
>>> os.listdir()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: 19
>>> 

```

**Comment by jerryneedell** [comment]
If I go back to master ad build feather52, the FS is created as expected and ampy also works normally

**Comment by arturo182** [comment]
Thanks for testing @jerryneedell , I'm having problems getting my feather 52 to work at all, so can't test just yet. But I did one change to the FLASH management code and I suspect it's something there, 19 is ENODEV https://github.com/adafruit/circuitpython/blob/master/py/mperrno.h#L57

**Comment by arturo182** [comment]
Most likely broken here https://github.com/adafruit/circuitpython/pull/601/commits/47c8e20e5a17b72078faa4b1962c515def819578

**Comment by microbuilder** [comment]
@arturo182 I really don't get why two boards don't show up at all over USB, even with the CP210x drivers installed (but even without them it should still show up on the USB bus). Have you tried using the J-Link (or CMSIS-DAP, etc.) to wipe the flash, and the reprogram the bootloader here: https://github.com/adafruit/circuitpython/tree/master/ports/nrf/boards/feather52/bootloader ... though the USB Serial bridge is a dedicated chip separate from the nRF52832??? I have about 15 of those boards, and haven't seen or heard of USB never enumerating with the CP2104, and the chance of two boards slipping through production tests is extremely small. :( Not sure what's happening though.

**Comment by arturo182** [comment]
I have tried JLink and that works, meaning the nRF is fine, the LEDs also work meaning power is delivered to the board. The FW doesn't matter since the CP2102 is dedicated so it should work anyway. I get no USB activity whatsoever on Linux and Windows, really strange.

**Comment by microbuilder** [comment]
Yeah, I just thought the UART pin state on the nRF52832 might be causing some sort of conflict with the CP2104 at startup, but I'm totally stretching my imagination here since we've sold thousands of these things and this is the first I've seen this, and on two boards at once. :(

**Comment by ladyada** [comment]
@arturo182 odd! we do check each one with a linux computer to program in the firmware over USB so its...unlikely you have two bad ones :) can you try a few different USB cables, those charge-only cables are a scourge.

**Comment by arturo182** [comment]
Ok, so this is super strange, out of 3 cables, I got the Feather working only with 1, and I know the other 2 have data wires, because I've been using them with the Nordic DKs. Maybe I can experiment a little more later to see if I can make sense of this. But now my Feathers enumerate and I get a tty! So I can soon test and fix the code and we can merge! :)

**Comment by ladyada** [comment]
Yay! I dont know what it is but there has been a lot of 'bad' usb cables. we're also going to tweak that nrf52 design to be low-power capable, might help as well. thanks for being resiliant :)

**Comment by arturo182** [comment]
Got my Feather working with the DFU bootloader and verified that all three cases (no SD, v2.0.1 and v5.0.0) work. Also fixed the flash issue @jerryneedell found.

**Comment by tannewt** [comment]
@microbuilder and @arturo182 is this ready to merge?

**Comment by arturo182** [comment]
Yes, pending @microbuilder's review.

**Pull request merged**